azurelinuxagent/agent.py

```diff
@@ -383,20 +380,17 @@ def parse_args(sys_args):
             cmd = AgentCommands.CollectLogs
         elif re.match(regex_cmd_format.format("full"), arg):
             log_collector_full_mode = True
-        elif re.match(regex_cmd_format.format(AgentCommands.SetupFirewall), arg):
-            cmd = AgentCommands.SetupFirewall
-        elif re.match(regex_cmd_format.format("dst_ip=(?P<dst_ip>[\\d.]{7,})"), arg):
-            firewall_metadata['dst_ip'] = re.match(regex_cmd_format.format("dst_ip=(?P<dst_ip>[\\d.]{7,})"), arg).group(
-                'dst_ip')
-        elif re.match(regex_cmd_format.format("uid=(?P<uid>[\\d]+)"), arg):
-            firewall_metadata['uid'] = re.match(regex_cmd_format.format("uid=(?P<uid>[\\d]+)"), arg).group('uid')
-        elif re.match(regex_cmd_format.format("(w|wait)$"), arg):
-            firewall_metadata['wait'] = "-w"
         else:
-            cmd = AgentCommands.Help
-            break
+            regex_cmd = regex_cmd_format.format("{0}=(?P<endpoint>[\\d.]{{7,}})".format(AgentCommands.SetupFirewall))
```

Comment:
now the command line takes only "-setup-firewall=<IP>"

Reply:
Makes sense, though I need the result of the match, so I do not think I can use an elif (unless I evaluate twice). Any suggestions?

Reply:
I see, maybe do the match before the first if condition. Although that may defeat the intent of making this more readable. If moving it creates confusion, i'm ok with it as is

Reply:
Yes, moving it to the top of the if does not help in terms of readability. On the other hand, evaluating twice is not such a big deal in this case, so if it helps for readability in your opinion, I can add it to the next iteration of this code (I still need to add unit test for the env thread)